app/Http/Controllers/ProductsController_SA.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Products_SA;
+use App\Helpers\ResponseHelper;
+use App\Http\Requests\ProductRequest_SA;
+use App\Http\Resources\ProductResource_SA;
+use App\Helpers\PaginationHelper;
+
+
+class ProductsController_SA extends Controller
+{
+    public function getData(ProductRequest_SA $request)
+    {
+        if (!$request->query()) {
+
+            if ($request->paginate) {
+
+                $currentPage = $request->input('page', 1);
+                $perPage = $request->input('pageSize', 5);
+
+                $products = Products_SA::paginate($perPage, ['*'], 'page', $currentPage);
+
+                $myData = PaginationHelper::paginateMyData($products, $currentPage, $perPage);
+
+                // if no query but pagination true
+                return response()->json(ResponseHelper::sendResponse(true, 200, "Data Fetched Successfully", $myData));
+            }
+
+            // If no query and no pagination
+            return response()->json(ResponseHelper::sendResponse(true, 200, "Data Fetched Successfully", ProductResource_SA::collection(Products_SA::all())));
+        }
+
+        $query = Products_SA::query();
+
+        // Apply filters
+        if ($request->has('name')) {
+            $query->where('name', 'LIKE', '%' . $request->input('name') . '%');
+        }
+
+        if ($request->has('startDate') && $request->has('endDate')) {
```

Comment:
please combine the date filters and avoid redundant checks

Reply:
Sir I did this because if I make different checks for start date and end date then there would be two queries if both start and end were provided. Therefore I add an extra condition to ensure that if both of them are available then we would find a product between them using "whereBetween" and this would take only single query instead of two.
And the first condition isn't satisfied then also only one query will be executed.

![image](https://github.com/ZeeshanHabib7/demo-internee-app-git/assets/119349274/696c851c-fd84-4991-b8b5-d3e211723b8f)